Dockerfile

```diff
@@ -1,47 +1,62 @@
+#Set mosquitto and plugin versions.
+#Change them as per your needs.
+ARG MOSQUITTO_VERSION=2.0.12
+ARG PLUGIN_VERSION=1.8.2
+
 #Use debian:stable-slim as a builder and then copy everything.
 FROM debian:stable-slim as builder
-
-#Set mosquitto and plugin versions.
-#Change them for your needs.
-ENV MOSQUITTO_VERSION=1.6.10
-ENV PLUGIN_VERSION=0.6.1
-ENV GO_VERSION=1.13.8
+ARG MOSQUITTO_VERSION
 
 WORKDIR /app
 
 #Get mosquitto build dependencies.
-RUN apt-get update && apt-get install -y libwebsockets8 libwebsockets-dev libc-ares2 libc-ares-dev openssl uuid uuid-dev wget build-essential git
+RUN apt update && apt install -y wget build-essential cmake libssl-dev  libcjson-dev libwebsockets-dev
 RUN mkdir -p mosquitto/auth mosquitto/conf.d
 
 RUN wget http://mosquitto.org/files/source/mosquitto-${MOSQUITTO_VERSION}.tar.gz
 RUN tar xzvf mosquitto-${MOSQUITTO_VERSION}.tar.gz && rm mosquitto-${MOSQUITTO_VERSION}.tar.gz 
 
 #Build mosquitto.
-RUN cd mosquitto-${MOSQUITTO_VERSION} && make WITH_WEBSOCKETS=yes && make install && cd ..
+RUN cd mosquitto-${MOSQUITTO_VERSION} && make CFLAGS="-Wall -O2 -I/build/lws/include" LDFLAGS="-L/build/lws/lib" WITH_WEBSOCKETS=yes && make install && cd ..
+
+# Use golang:latest as a builder for the Mosquitto Go Auth plugin.
+FROM golang:latest AS go_auth_builder
```

Comment:
i would prefer a fixed golang version for the future.

Reply:
Okay , I had some issues using the wget to install go so I used the strategy used in iegomez Dockerfile. I can work on this to get it running using your way.